articles/150_ros_command_line.md

```diff
@@ -0,0 +1,107 @@
+- Suitable for tab completion.
+
+Unfortunately, additional devices must be put in place as these coexist with user-defined options -- part of the rationale behind ROS 1 command line design.
+
+## Namespacing
+
+To prevent ROS specific command line options from interacting with user-defined ones, the former are scoped using the `--ros-args` option and a trailing double dash token (`--`):
+
+```sh
+ros2 run some_package my_node [<user-defined-pre-arg0>...] \
+  --ros-args <ros-specific-arg-0> [<ros-specific-arg-1>...] -- <user-defined-arg-1> [<user-defined-arg-2>...]
+```
+
+If no user defined arguments are provided after ROS specific arguments are, the double dash token (`--`) may be elided:
+
+```sh
+ros2 run some_package my_node [<user-defined-arg-0>...] --ros-args <ros-specific-arg-0> [<ros-specific-arg-1>...]
+```
+
+## Capabilities
+
+### Summary
+
+As a quick summary of ROS command line capabilities:
+
+- For name remapping, use either `--remap from:=to` or `-r from:=to`.
+- For parameter assignment, use either `--param name:=value` or `-p name:=value`.
+
+### Name remapping rules
+
+Remapping rules may be introduced using the `--remap`/`-r` option.
+This option takes a single `from:=to` remapping rule.
+
+As an example, to remap from `foo` to `bar` upon running `some_node`, one may execute:
+
+```sh
+ros2 run some_package some_node --ros-args --remap foo:=bar
+```
+
+or its shorter equivalent:
+
+```sh
+ros2 run some_package some_node --ros-args -r foo:=bar
+```
+
+### Parameter assignment statements
+
+Parameter assignment may be achieved using the `--param`/`-p` option.
+This option takes a single `name:=value` assignment statement, where `value` is in YAML format.
+
+As an example, to assign an integer value of `2` to an `int_param` upon running `some_node`, one may execute:
+
+```sh
+ros2 run some_package some_node --ros-args --param int_param:=2
+```
+
+or its shorter equivalent:
+
+```sh
+ros2 run some_package some_node --ros-args -p int_param:=2
+```
+
+## Alternative designs
+
+Other, alternative designs were under discussion.
+
+### Additional operators
+
+Stop using the same `:=` operator for parameter assignments and name remapping rules and introduce additional operators e.g. `:=` for parameter assignment and `~=` for name remapping. This keeps the command line verbosity at a minimum, but is error prone.
+
+### Full name addressing
+
+Rely on full name addressing to disambiguate operator significance e.g. `rosparam://this:=that` would mean to assign a `that` string value to parameter `this` while `rostopic://this:=that` would mean to remap topic `this` to `that`. This signficantly increases the command line verbosity.
+
+### Prefixed option names
+
+Remove the need for double dash tokens (`--`), conventionally used to signify the end of CLI options for a command, by adding the `--ros-` prefix to all ROS specific command line options e.g. `--ros-remap`, `--ros-param`, etc. This somewhat increases the command line verbosity.
```

Comment:
I would also say that this one makes the extraction of the ROS arguments slightly more difficult because you need to know about all ros options ahead of time, whereas the currently proposal will always be removable with just two or three rules to consider.

Reply:
True that. See beaf7bec3dfc45db907f8c89e30801dcfdc3eba1.